src/ucp/core/ucp_request.inl

```diff
@@ -619,34 +619,38 @@ ucp_request_init_dt_reg_from_memh(ucp_request_t *req, ucp_md_map_t md_map,
     }
 }
 
-static UCS_F_ALWAYS_INLINE ucs_status_t
-ucp_request_check_user_memh(ucp_request_t *req, ucp_mem_h memh, void *buffer,
-                            size_t length, ucs_memory_type_t mem_type)
+/* Returns whether user-provided memory handle can be used. If the result is 0,
+   *status_p is set to the error code. */
+static UCS_F_ALWAYS_INLINE int
+ucp_request_is_user_memh_valid(ucp_request_t *req,
+                               const ucp_request_param_t *param, void *buffer,
+                               size_t length, ucp_datatype_t datatype,
+                               ucs_memory_type_t mem_type,
+                               ucs_status_t *status_p)
 {
+    /* User-provided memh supported only on contig type with proto_v1 */
```

Comment:
where do we check proto version?

Reply:
proto v2 should not call this function. added assert to document it